core/consensus/babe/src/lib.rs

```diff
@@ -418,19 +425,24 @@ macro_rules! babe_err {
 	};
 }
 
-fn find_pre_digest<B: Block>(header: &B::Header) -> Result<BabePreDigest, String>
+fn find_pre_digest<B: Block>(header: &B::Header) -> Result<(Option<Epoch>, BabePreDigest), String>
 	where DigestItemFor<B>: CompatibleDigestItem,
 {
-	let mut pre_digest: Option<_> = None;
+	let (mut pre_digest, mut epoch): (Option<_>, Option<_>) = (None, None);
 	for log in header.digest().logs() {
 		trace!(target: "babe", "Checking log {:?}", log);
 		match (log.as_babe_pre_digest(), pre_digest.is_some()) {
 			(Some(_), true) => Err(babe_err!("Multiple BABE pre-runtime headers, rejecting!"))?,
 			(None, _) => trace!(target: "babe", "Ignoring digest not meant for us"),
 			(s, false) => pre_digest = s,
 		}
+		match (log.as_babe_epoch(), epoch.is_some()) {
```

Comment:
As we are replicating the same code as above, this could be extracted into its own function/closure :woman_shrugging: